security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -116,12 +118,15 @@ func (p Plugin) ExchangeToken(ctx context.Context, trustDomain, k8sSAjwt string)
 	return respData.AccessToken, time.Now().Add(time.Second * time.Duration(respData.ExpiresIn)), resp.StatusCode, nil
 }
 
-func constructAudience(trustDomain string) string {
-	if GKEClusterURL == "" {
-		return trustDomain
+func constructAudience(platform, trustDomain string) string {
+	switch platform {
+	case security.GCE:
```

Comment:
Why are we leaking GCE code into non gce part of the code?

Reply:
Why? STS is not a google thing

Reply:
Technically this is GCP specific code as well ( it's for the google provider ), so should be moved in the same package with all the other google-platform specific code.